src/main/scala/cromwell/engine/backend/io/package.scala

```diff
@@ -0,0 +1,58 @@
+package cromwell.engine.backend
+
+import java.nio.file._
+
+import cromwell.engine.backend.io.filesystem.gcs.{ContentTypeOption, GcsFileSystem, GcsFileSystemProvider, NioGcsPath}
+
+import scala.util.{Success, Try}
+
+package object io {
+  object PathString {
+    val defaultGCSFileSystem = GcsFileSystem.defaultGcsFileSystem
+    val defaultFileSystem = FileSystems.getDefault
+    private val defaultFileSystemsList = List(defaultGCSFileSystem, defaultFileSystem)
+
+    implicit class UriString(val str: String) extends AnyVal {
+
+      def isGcsUrl: Boolean = str.startsWith("gs://")
+      def isUriWithProtocol: Boolean = "^[a-z]+://".r.findFirstIn(str).nonEmpty
+
+      def toPath(implicit fss: List[FileSystem] = defaultFileSystemsList): Path = {
```

Comment:
Well, I'm able to follow what's going on, and I guess ultimately that's what matters most.  If I were writing this, I'd probably try to see if it can be done cleanly without the implicit parameter... but I'm just one person who has a bias :)